### PR TITLE
3436 - Refactor isinviewport and fix dropdown closing

### LINKED ISSUE #3436: General: Combine isInViewport to a shared utils
**Is your feature request related to a problem? Please describe.**

Found two occurances of isInViewport. Also some folk would like this util to use. So extract it into a util and refactor the code

**Describe the solution you'd like**

- [ ]  Take isInViewport out of popupmenu
- [ ]  Take isInViewport out of datagrid
- [ ]  Put it in utils

**Describe alternatives you've considered**
Leave it

**Additional context**
Ms Teams Chat

### PR DESCRIPTION
**Pull request by tmcconechy**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Exposed a duplicate function that was in two places in shared public utils. Removed some pages from the utils section on the docs site that had no document pages. Fixed scrolling issue on dropdown noticed when testing the refactor.

**Related github/jira issue (required)**:
Fixes #3436

**Steps necessary to review your pull request (required)**:
- in the console try `Soho.utils.isInViewPort(elem)` 
- go to http://localhost:4000/components/datagrid/example-filter.html
- size the page so there is a horizontal scroll
- open the dropdown in the filter row
- scroll -> it should close
- go to http://localhost:4000/components/dropdown/test-on-modal.html
- open the dropdown
- scroll -> it should close
- go to http://localhost:4000/components/dropdown/test-mobile-scrolling.html
- scroll down in the card to the dropdown
- open the dropdown
- scroll the card -> it should close

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Comment by ericangeles** [comment]
Just noticed a minor issue on the tooltip.

- Go to http://localhost:4000/components/dropdown/test-on-modal.html
- Scroll to the bottom
- Open the `Add Context` button
- Select the longest text in the dropdown
- Hover the dropdown
- Tooltip is being cutoff

tooltip

The rest are good! 🍌

**Comment by tmcconechy** [comment]
@ericangeles this is a seperate issue https://github.com/infor-design/enterprise/issues/3185 i have that scheduled soon (next sprint).

File: src/components/dropdown/dropdown.js [hunk 1]
```diff
@@ -1978,7 +1978,7 @@ Dropdown.prototype = {
     // When the Dropdown is located within a scrollable section,
     // the dropdown must close if that section is scrolled.
     let parentScrollableArea = $('.modal.is-visible .modal-body-wrapper');
-    const subScrollableSection = self.element.closest('.scrollable, .scrollable-x, .scrollable-y', '.datagrid-wrapper');
+    const subScrollableSection = self.element.closest('.scrollable, .scrollable-x, .scrollable-y, .card-content');
```
Comment (EdwardCoyle) [inline src/components/dropdown/dropdown.js:1981]: Maybe in order to fix that, it needs to find all scrollable parents?

File: src/utils/utils.js [hunk 2]
```diff
@@ -1122,4 +1122,19 @@ utils.deepCopy = function (arrayOrObject) {
   return copy(arrayOrObject);
 };
 
+/**
+ * Check if given element is within the viewport.
+ * @private
+ * @param {object} element The element to check
+ * @returns {boolean} Whether or not the element is in the viewport.
+ */
+utils.isInViewport = function isInViewport(element) {
```
Comment (EdwardCoyle) [inline src/utils/utils.js:1131]: 💯 We can make great use of this

**Comment by tmcconechy** [comment]
@EdwardCoyle @deep7102 @ericangeles 
This is ready to review. Again. I fixed the issue @EdwardCoyle mentioned. It will now shut on either scrollbar being scrolled.

**Pull request merged**